doc/automaatiotestit.md

```diff
@@ -0,0 +1,215 @@
+
+Geoviitteen yksikkötestit sijaitsevat tietylle toiminnallisuudelle varatussa tiedostohierarkiassa, ja testikoodin
+polkurakenne vastaa varsinaisen logiikan polkurakennetta. Kun varsinainen logiikka löytyy esimerkiksi polusta
+
+```
+infra/src/main/kotlin/fi/fta/geoviite/infra/geometry
+```
+
+niin geometry-pakettiin liittyvä yksikkötestikoodi löytyy vastaavasti polusta
+
+```
+infra/src/test/kotlin/fi/fta/geoviite/infra/geometry
+```
+
+joiden ainut ero on kansio 'main|test' polun alkupuolella.
+
+Yksikkötestitiedostojen nimet noudattavat rakennetta `<TestattavaOminaisuusTiedosto>Test.kt`, eli tiedoston nimi alkaa
+main-puolelta löytyvällä tiedostonimellä. Vastaavasti yksikkötestitiedosto taas päättyy `Test`-loppuliitteellä.
+
+Siispä esimerkiksi varsinaista logiikkaa sisältävän tiedoston
+
+```
+infra/src/main/kotlin/fi/fta/geoviite/infra/geometry/ElementListing.kt
+```
+
+yksikkötestit löytyvät tiedostosta
+
+```
+infra/src/test/kotlin/fi/fta/geoviite/infra/geometry/ElementListingTest.kt
+```
+
+# Integraatiotestit (integration tests)
+
+Integraatiotestit voivat hyödyntää useampaa kuin yhtä prosessia (säiettä). Tämä tarkoittaa, että ne voivat hyödyntää
+esimerkiksi tietokantaa taikka ulkopuolisia palveluita. Geoviitteen integraatiotestit tarkoittavat käytännössä kuitenkin
+oikeaa tietokantaprosessia hyödyntäviä testejä. Geoviitteen integraatiotestit eivät siis lähtökohtaisesti käytä
+valepalveluja (mock service), kuten valetietokantaa, lukuunottamatta ulkoisiin integraatioihin (esim. Ratko,
+ProjektiVelho) muodostettuja integraatiotestejä. Geoviitteen integraatiotestien aikana ei siis kutsuta esimerkiksi
+kolmannen osapuolen rajapintoja.
+
+Geoviitteen integraatiotestit ovat lähtökohtaisesti muodostettu DAO (data access object) sekä palvelu (Service)
+tasoille.
+DAO-testejä ei voida toteuttaa yksikkötesteinä, sillä ne hyödyntävät tietokantaa. Vastaavasti palvelut hyödyntävät
+tyypillisesti logiikassaan tietokannan dataa oman DAO:nsa taikka jonkin toisen palvelun kautta, joten palveluidenkaan
+testejä ei lähtökohtaisesti voida toteuttaa yksikkötesteinä.
+
+Esimerkkinä integraatiotestistä voidaan pitää esimerkiksi tietyn tietorakenteen hakemista oikein tietokannasta DAO:n
+avulla.
+
+## Integraatiotestien sijainti ja nimitys
+
+Integraatiotestien nimet noudattavat rakennetta `<TestattavaOminaisuusTiedosto>IT.kt`, eli tiedoston nimi alkaa
+main-puolelta löytyvällä tiedostonimellä. Vastaavasti testitiedosto taas päättyy `IT`-loppuliitteellä. Integraatiotestit
+sijaitsevat yksikkötestien tavoin lähes vastaavassa tiedostopolussa, kuin varsinainen testattava logiikka.
+
+Kun esimerkiksi varsinaista testattavaa logiikkaa sisältävä tiedosto on
+
+```
+infra/src/main/kotlin/fi/fta/geoviite/infra/geometry/GeometryService.kt
+```
+
+niin sen integraatiotestit löytyvät tiedostosta
+
+```
+infra/src/test/kotlin/fi/fta/geoviite/infra/geometry/GeometryServiceIT.kt
+```
+
+jossa kannattaa huomioida jälleen polun alussa ero "main|test"-kansion välillä.
```

Comment:
Koska tuo `@SpringBootTest` on tosiaan IT- ja E2E-testispesifistä juttua niin lisäisin tänne siitä maininnan